src/sonic-dhcp-utilities/tests/test_data/kea-dhcp4.conf.j2

```diff
@@ -63,10 +63,12 @@
                         "always-send": {{ config["always_send"] }}
                     },
     {%- endfor %}
+    {%- if "gateway" in subnet_info %}
                     {
                         "name": "routers",
-                        "data": "{{ subnet_info["gateway"] if "gateway" in subnet_info else subnet_info["server_id"] }}"
+                        "data": "{{ subnet_info["gateway"] }}"
```

Comment:
When it could be `subnet_info["server_id"]`? Seems like new code lost that feature

Reply:
In lin 72 to line 75 of this file. It's for clients to know which server is offering the IP address lease. For now we treated as non-configurable and always use DHCP-interface ip (Generally, it's vlan interface ip)

Reply:
Why we trying to configure `subnet_info["server_id"]` in routers entry previously?

Reply:
It's a invalid condition. In previous,gateway would always appear in dhcp_config and subnet_obj would always have "gateway", which would be pass to j2 template. It means that router always was configured as gateway rather than "server_id" in previous.
![S50126-16461381_mark via](https://github.com/user-attachments/assets/42bcf296-ae9a-4988-9611-ec5d2827818c)